File: cpp/Platform.Data.Tests/HybridTests.cpp
namespace Platform::Data::Tests
{
    TEST(HybridTests, ObjectConstructorTest)
    {
        ASSERT_EQ(0, Hybrid(std::uint8_t{128}).AbsoluteValue());
        ASSERT_EQ(1, Hybrid(static_cast<uint8_t>(-1)).AbsoluteValue());
        ASSERT_EQ(0, Hybrid(std::uint8_t{0}).AbsoluteValue());
        ASSERT_EQ(1, Hybrid(std::uint8_t{1}).AbsoluteValue());
    };
}

File: csharp/Platform.Data.Tests/HybridTests.cs
using Xunit;

namespace Platform.Data.Tests
{
    /// <summary>
    /// <para>
    /// Represents the hybrid tests.
    /// </para>
    /// <para></para>
    /// </summary>
    public static class HybridTests
    {
        /// <summary>
        /// <para>
        /// Tests that object constructor test.
        /// </para>
        /// <para></para>
        /// </summary>
        [Fact]
        public static void ObjectConstructorTest()
        {
            Assert.Equal(0, new Hybrid<byte>(unchecked((byte)128)).AbsoluteValue);
            Assert.Equal(0, new Hybrid<byte>((object)128).AbsoluteValue);
            Assert.Equal(1, new Hybrid<byte>(unchecked((byte)-1)).AbsoluteValue);
            Assert.Equal(1, new Hybrid<byte>((object)-1).AbsoluteValue);
            Assert.Equal(0, new Hybrid<byte>(unchecked((byte)0)).AbsoluteValue);
            Assert.Equal(0, new Hybrid<byte>((object)0).AbsoluteValue);
            Assert.Equal(1, new Hybrid<byte>(unchecked((byte)1)).AbsoluteValue);
            Assert.Equal(1, new Hybrid<byte>((object)1).AbsoluteValue);
        }
    }
}
